### PR TITLE
Ganglia

### PR DESCRIPTION
**Pull request by zmcgrew**
This adds two new aports: ganglia and ganglia-web. Ganglia is used for monitoring large environments, such as HPC nodes. Ganglia-web is a php frontend that can display and leverage rrdtool to generate graphs from all of the data.

### CODE REVIEW

File: community/ganglia-web/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Contributor: Zach McGrew <zmcgrew@gmail.com>
+# Maintainer:
+pkgname="ganglia-web"
+pkgver="3.7.5"
+pkgrel=0
+pkgdesc="Web visualisation for Ganglia monitoring tool"
+url="https://github.com/ganglia/ganglia-web/wiki"
+arch="noarch"
+license="BSD"
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia-web/APKBUILD]: Replaced with BSD-3-Clause

File: community/ganglia-web/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+depends="ganglia"
+makedepends="rsync"
+# Package just copies a directory to specified location -- No tests
+options="!check"
+install="$pkgname.pre-install $pkgname.post-install"
+pkggroups="www-data"
+subpackages=""
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia-web/APKBUILD]: Removed subpackages line.

File: community/ganglia-web/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+source="$pkgname-$pkgver.tar.gz::https://github.com/ganglia/ganglia-web/archive/$pkgver.tar.gz"
+builddir="$srcdir/ganglia-web-$pkgver"
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia-web/APKBUILD]: Removed builddir line.

File: community/ganglia-web/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+package() {
+	#	cd "$srcdir/$pkgname-$pkgver" || exit 1
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia-web/APKBUILD]: Comment and extra line removed.

File: community/ganglia/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+# Maintainer:
+pkgname="ganglia"
+pkgver="3.7.2"
+pkgrel=0
+pkgdesc="Ganglia is a scalable distributed monitoring system for high-performance computing systems"
+pkggroups="ganglia"
+pkgusers="ganglia"
+url="http://ganglia.info/"
+arch="all"
+license="BSD"
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia/APKBUILD]: Replaced with BSD-3-Clause.

File: community/ganglia/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	gmetad.initd
+	riemann.proto::https://raw.githubusercontent.com/aphyr/riemann-java-client/2.5.0/src/main/proto/riemann/proto.proto
+	uid.patch
+	runstatedir.patch
+	"
+builddir="$srcdir/monitor-core-$pkgver"
+
+prepare() {
+	cd "$builddir"
+	./bootstrap || return 1
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia/APKBUILD]: Removed || return 1

File: community/ganglia/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	gmond.initd
+	gmetad.initd
+	riemann.proto::https://raw.githubusercontent.com/aphyr/riemann-java-client/2.5.0/src/main/proto/riemann/proto.proto
+	uid.patch
+	runstatedir.patch
+	"
+builddir="$srcdir/monitor-core-$pkgver"
+
+prepare() {
+	cd "$builddir"
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia/APKBUILD]: Also removed directory switch to $builddir

File: community/ganglia/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+	./bootstrap || return 1
+	default_prepare
+
+	cd "$srcdir"
+	protoc-c --c_out="$builddir/gmetad" riemann.proto
+}
+
+build() {
+	cd "$builddir"
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia/APKBUILD]: Removed unneeded directory change

File: community/ganglia/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+		--with-python=/usr/bin/python2
+	make
+}
+
+check() {
+	make check
+}
+
+package() {
+	cd "$builddir"
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia/APKBUILD]: Removed unneeded directory change

File: community/ganglia-web/ganglia-web.post-install [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+echo -e "Make sure that your php open_basedir includes\n" \
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia-web/ganglia-web.post-install]: Replaced with here-doc (cat <<EOF)

File: community/ganglia/APKBUILD [hunk 11]
```diff
@@ -0,0 +1,104 @@
+# Contributor: Zach McGrew <zmcgrew@gmail.com>
+# Maintainer:
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia/APKBUILD]: I was told in #alpine-dev that a maintainer is not required. While I do plan to use this package and wanted to share it with the community, I do not intend to watch for new releases as releases tend to be infrequent.

File: community/ganglia-web/APKBUILD [hunk 12]
```diff
@@ -0,0 +1,39 @@
+# Contributor: Zach McGrew <zmcgrew@gmail.com>
+# Maintainer:
```
[the remark this answers is not in the document]
Comment (zmcgrew) [reply community/ganglia-web/APKBUILD]: I was told in #alpine-dev that a maintainer is not required. While I do plan to use this package and wanted to share it with the community, I do not intend to watch for new releases as releases tend to be infrequent.

File: community/ganglia/APKBUILD (hunk 11 above)
[the remark this answers is not in the document]
Comment (maxice8) [reply community/ganglia/APKBUILD]: fair

**Comment by zmcgrew** [comment]
> Packages should start in testing/ not community/

Not sure if this was noticed, but I moved them to testing/ as part of my changes.

**Comment by maxice8** [comment]
Yes i noticed that

**Pull request closed without merging**